blueman/main/Tray.py

```diff
@@ -0,0 +1,54 @@
+from importlib import import_module
+
+import logging
+
+from blueman.Functions import check_single_instance
+from blueman.main.AppletService import AppletService
+
+import gi
+gi.require_version('Gtk', '3.0')
+from gi.repository import GLib, Gio
+
+
+class BluemanTray(object):
+    def __init__(self):
+        check_single_instance("blueman-tray")
+
+        logging.basicConfig(level=logging.INFO)
+
+        applet = AppletService()
+
+        main_loop = GLib.MainLoop()
+
+        Gio.bus_watch_name(Gio.BusType.SESSION, 'org.blueman.Applet',
+                           Gio.BusNameWatcherFlags.NONE, None, main_loop.quit)
+
+        indicator_name = applet.GetStatusIconImplementation()
+        indicator_name = 'GtkStatusIcon'
+        logging.info('Using indicator "%s"' % indicator_name)
+        indicator_class = getattr(import_module('blueman.main.indicators.' + indicator_name), indicator_name)
+        self.indicator = indicator_class(applet.GetIconName(), self._activate_menu_item, self._activate_status_icon)
+
+        applet.connect('g-signal', self.on_signal)
+
+        self.indicator.set_text(applet.GetText())
+        self.indicator.set_visibility(applet.GetVisibility())
+        self.indicator.set_menu(applet.GetMenu())
+
+        main_loop.run()
+
+    def _activate_menu_item(self, *indexes):
+        return AppletService().ActivateMenuItem('(ai)', indexes)
+
+    def _activate_status_icon(self):
+        return AppletService().Activate()
+
+    def on_signal(self, _applet, sender_name, signal_name, args):
+        if signal_name == 'IconNameChanged':
+            self.indicator.set_icon(*args)
+        elif signal_name == 'TextChanged':
+            self.indicator.set_text(*args)
+        elif signal_name == 'VisibilityChanged':
+            self.indicator.set_visibility(*args)
+        elif signal_name == 'MenuChanged':
+            self.indicator.set_menu(*args)
```

Comment:
The reason the exititem is not changed is because the signal MenuChanged is never receieved here. I do see others like ``VisibilityChanged`` and ``VisibilityChanged`` so it appears to be on the dbus service side.